trin-core/src/jsonrpc/types.rs

```diff
@@ -101,14 +104,56 @@ impl TryFrom<&Value> for PingParams {
     fn try_from(param: &Value) -> Result<Self, Self::Error> {
         match param.as_str() {
             Some(val) => match Enr::from_str(val) {
-                Ok(val) => Ok(PingParams { enr: val }),
+                Ok(enr) => Ok(Self { enr }),
                 Err(_) => Err(ValidationError::new("Invalid enr param")),
             },
             None => Err(ValidationError::new("Missing or empty enr param")),
         }
     }
 }
 
+pub struct FindContentParams {
+    pub enr: Enr,
+    pub content_key: ByteList,
+}
+
+impl TryFrom<Params> for FindContentParams {
+    type Error = ValidationError;
+
+    fn try_from(params: Params) -> Result<Self, Self::Error> {
+        match params {
+            Params::Array(val) => match val.len() {
+                2 => FindContentParams::try_from([&val[0], &val[1]]),
```

Comment:
Feels like there ought to be a cleaner way to do this. I believe [this post](https://stackoverflow.com/a/29570662/8412986) is suggesting that you could do: 
```suggestion
                2 => FindContentParams::try_from(val.try_into().unwrap()),
```
Hm, maybe that's not any better...